File: src/scene/VictoryScene.java
package scene;

import org.newdawn.slick.Color;
import org.newdawn.slick.Font;
import org.newdawn.slick.GameContainer;
import org.newdawn.slick.SlickException;
import org.newdawn.slick.state.StateBasedGame;

import component.Label;
import component.Panel;
import component.Positionable;
import core.FontStore;

/**
 * A scene displayed upon reaching the final destination along the trail.
 */
public class VictoryScene extends Scene {
	public static final SceneID ID = SceneID.VICTORY;
	
	@Override
	public void init(GameContainer container, StateBasedGame game) throws SlickException {
		super.init(container, game);
		
		Font h1 = FontStore.get().getFont(FontStore.FontID.H1);
		Label titleLabel = new Label(container, h1, Color.green, "Victory!");
		mainLayer.add(titleLabel, mainLayer.getPosition(Positionable.ReferencePoint.CENTERCENTER), Positionable.ReferencePoint.CENTERCENTER);
		
		backgroundLayer.add(new Panel(container, Color.black));
	}
	
	@Override
	public void update(GameContainer container, StateBasedGame game, int delta) throws SlickException {
		return;
	}

	@Override
	public int getID() {
		return ID.ordinal();
	}
}

File: web/src/scene/VictoryScene.ts
import { Scene } from './Scene';
import { SceneID } from './SceneID';
import { Panel } from '../component/Panel';
import { Label } from '../component/Label';
import { ReferencePoint } from '../component/Component';
import { Color } from '../core/Color';
import { FontStore, FontID } from '../core/FontStore';

export class VictoryScene extends Scene {
  constructor(canvasWidth: number, canvasHeight: number) {
    super();
    this.init(canvasWidth, canvasHeight);
  }

  private init(canvasWidth: number, canvasHeight: number): void {
    const background = new Panel(canvasWidth, canvasHeight, Color.black);
    this.backgroundLayer.add(background);
    background.setPosition({ x: 0, y: 0 }, ReferencePoint.TOPLEFT);

    const h1Font = FontStore.getFont(FontID.H1);
    const victoryLabel = new Label("Victory!", h1Font, Color.green, canvasWidth);
    this.mainLayer.add(victoryLabel);
    victoryLabel.setPosition(
      { x: canvasWidth / 2, y: canvasHeight / 2 },
      ReferencePoint.CENTERCENTER
    );
  }

  getID(): SceneID {
    return SceneID.VICTORY;
  }
}
